stdlib/stdlib.mli

```diff
@@ -51,6 +58,68 @@ exception Exit
 (** The [Exit] exception is not raised by any library function.  It is
     provided for use in your programs. *)
 
+exception Match_failure of (string * int * int)
+(** Exception raised when none of the cases of a pattern-matching
+   apply. The arguments are the location of the match keyword in the
+   source code (file name, line number, column number). *)
+
```

Comment:
Unfortunately, I suspect this is not quite enough: predef.ml adds these attributes [@ocaml.warn_on_literal_pattern] on exceptions, and given that we shadow the built-in ones, we probably break the warnings.

Reply:
(and if that's correct, there's the same problem in the other stdlib, though in the .ml since it doesn't have a .mli)

Reply:
Ping @yallop for the comment above.

Reply:
@sliquister is correct -- omitting the `[@ocaml.warn_on_literal_pattern]` attribute does break the warning -- so I've added the attribute to the exceptions with constructors.